amm/contracts/stable_pool/lib.rs

```diff
@@ -287,6 +287,9 @@ pub mod stable_pool {
             let current_time = self.env().block_timestamp();
             let mut rate_changed = false;
             for rate in self.pool.token_rates.iter_mut() {
+                // [audit] then we first update rate successfully, then rate changed
+                // will be set to true and no further rate.update_rate calls will be done in this loop.
+                // Not sure if this is what we meant - maybe bitwise `or` operator (`|`) should be used?
```

Comment:
Good catch! :raised_hands:

Reply:
nice, I would suggest not using bit operators to fix it, but some simple `if else`